### ISSUE TITLE
[Temporary fix] for Expected quality not found! error

### ISSUE THREAD

**Issue by IronSuit** [nishad/udemy-dl#214]
Hello everyone,

This fix is made possible by @smeggingsmegger to solve the Expected quality not found issue.
It took me nearly 5 hours to figure this method, this tutorial for people like me who don't know anything in regards to Python or any coding skills.

1. download python 2.7 from [https://www.python.org/downloads/ ](url)
2. Install python 2.7 but don't change the install directory and leave it as it is which is "C:\Python27\"
3. download and install "setuptools-0.6c11.win32-py2.7.exe" from [https://pypi.python.org/pypi/setuptools/0.6c11](url)
4. Open cmd and type the following `pushd C:\Python27\Scripts`
5. type the following command to install pip `easy_install pip`
6. download the git from [https://git-scm.com/download/win](url)
7. close and reopen cmd and type the following `pushd C:\Python27\Scripts`
8. Type `pip install -U git+https://github.com/smeggingsmegger/udemy-dl-windows.git@master` to download the fix
9. go to "C:\Python27" and create a new batch and name it `udemy-dl.bat` and write inside it 
```
@echo off
for /f %%p in ('where python.exe') do set pythonpath=%%p
set "_python=%pythonpath%"
for %%a in ("%_python%") do set "python_scripts=%%~dpa\scripts"
python.exe %python_scripts%\udemy-dl %*
```

step 9 is credited to @mrswadge

10. in cmd type the following `pushd C:\Python27`
11. now everything should be working as usual, so to download just type as normal `udemy-dl COURSELINK`

Please note that the above method worked for me and i **cannot guarantee** that it will work with everyone, but hey it's better than nothing.
Also this method is for Microsoft windows.

**Comment by fyz** [comment]
If anyone have still problem, for example with this error:
`Command "python setup.py egg_info" failed with error code 1 in "....." `

Solution, change:
`pip install -U git+https://github.com/smeggingsmegger/udemy-dl-windows.git@master` 
to 
`pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master`

For me working with builtin downloader, thanks

**Comment by KoushikKumar** [comment]
Thanks a lot. Its working fine!!

**Comment by blackstok2** [comment]
Help :/ 

'C:\Python27' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.
python.exe: can't open file '\scripts\udemy-dl': [Errno 2] No such file or directory

**Comment by alexcat25** [comment]
Hi blackstok2, you should configure the python to the PATH.
C:\Users\YourUser\AppData\Local\Programs\Python\Python36-32\Scripts;
C:\Users\YourUser\AppData\Local\Programs\Python\Python36-32;
C:\Python27;
C:\Python27\Scripts

Those folders should appear into the environment variables. Good luck.

**Comment by mailtoajay** [comment]
Hi,
I am getting this error when I am trying to download a course -
udemy-dl https://www.udemy.com/install-storefront-wordpress-theme-integrate-woocommerce/

[ERROR-61] Exception
Traceback (most recent call last):
  File "C:\Python27\\scripts\udemy-dl", line 7, in <module>
    main()
  File "C:\Python27\lib\site-packages\udemy_dl\udemy_dl.py", line 930, in main
    password = getpass.getpass(prompt='Password : ')
  File "C:\Python27\lib\getpass.py", line 95, in win_getpass
    msvcrt.putch(c)
TypeError: putch() argument 1 must be char, not unicode

**Comment by fyz** [comment]
**mailtoajay** solution for you:
`udemy-dl -u USERNAME -p PASSWORD link`
Read usage section
[https://github.com/nishad/udemy-dl#usage](url)

**Comment by noph34r** [comment]
my error is this

Traceback (most recent call last):
  File "C:\Python27\Lib\site-packages\udemy_dl\udemy_dl.py", line 33, in <module>
    from .download import download, DLException, USER_AGENT
ValueError: Attempted relative import in non-package

any help

**Comment by mailtoajay** [comment]
Thank You,  fyz. Now its working.

**Comment by yassinali** [comment]
Hi, iam still getting the error, i runned "udemy-dl pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master" and then i runned the link to download, but the error still existing. Please help me solve this issue.

**Comment by fyz** [comment]
**noph34r** try reinstall
**yassinali** bad command, read carefully steps and remove udemy-dl, only pip....

**Comment by yassinali** [comment]
Hi, iam getting error, it says SyntaxError: invalid syntax.

Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "copyright", "credits" or "license()" for more information.
>>> pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master
SyntaxError: invalid syntax
>>> 

please help me solve this issue.

**Comment by cnmoro** [comment]
wow. this worked perfectly.
thanks everyone for the support provided here.

**Comment by CillianCritchley** [comment]
worked great, thanks to all involved

**Comment by yassinali** [comment]
Can, anyony help me please.

**Comment by CillianCritchley** [comment]
run "pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master" not "udemy-dl pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master"

**Comment by yassinali** [comment]
when i run i get this error below:
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "copyright", "credits" or "license()" for more information.

pip install -U git+https://github.com/smeggingsmegger/udemy-dl.git@master
SyntaxError: invalid syntax

**Comment by CillianCritchley** [comment]
oh i see your post from last week now. I've no idea it works fine for me in linux, sorry.

**Comment by yassinali** [comment]
Iam trying to follow the steps again. When iam done i will tell.

**Comment by nishad** [comment]
Duplicate of https://github.com/nishad/udemy-dl/issues/207, Fixed with 133e73ac56ed3bcefdaf66badf8402d2bc89b0f5
